turbopack/crates/turbopack-ecmascript/src/side_effect_optimization/facade/module.rs

```diff
@@ -450,3 +456,36 @@ impl ChunkableModule for EcmascriptModuleFacadeModule {
 
 #[turbo_tasks::value_impl]
 impl EvaluatableAsset for EcmascriptModuleFacadeModule {}
+
+#[turbo_tasks::value_impl]
+impl MergeableModule for EcmascriptModuleFacadeModule {
+    #[turbo_tasks::function]
+    async fn merge(
+        self: Vc<Self>,
+        modules: Vc<MergeableModulesExposed>,
+        entry_points: Vc<MergeableModules>,
+    ) -> Result<Vc<Box<dyn ChunkableModule>>> {
+        Ok(Vc::upcast(*MergedEcmascriptModule::new(
+            modules
```

Comment:
create a new factory method on `MergedEcmascriptModule` that does these side-casts? it looks like all callers are doing something very similar